lib/transition/import/cloudfront_access_log_parser.rb

```diff
@@ -0,0 +1,60 @@
+require "ip"
+require "time"
+
+module Transition
+  module Import
+    class CloudfrontAccessLogParser
+      def self.fields
+        @fields || %i[date time x-edge-location sc-bytes c-ip cs-method cs(Host) cs-uri-stem sc-status cs(Referer) cs(User-Agent) cs-uri-query cs(Cookie) x-edge-result-type x-edge-request-id x-host-header cs-protocol cs-bytes time-taken x-forwarded-for ssl-protocol ssl-cipher x-edge-response-result-type cs-protocol-version fle-status fle-encrypted-fields c-port time-to-first-byte x-edge-detailed-result-type sc-content-type sc-content-len sc-range-start sc-range-end]
+      end
+      
+      class Entry < OpenStruct
+        def self.from_string(line)
+          mapping = {}
+          x = nil
+          line.scrub.split(/^(\d{4}-\d{2}-\d{2})\s(\d{2}:\d{2}:\d{2})\s(\w+-?\w+)\s(\d+)\s(\S+)\s(\w+)\s(\S+)\s(\S+)\s(\d+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\w+)\s(\S+)\s(\S+)\s(\w+)\s(\d+)\s(\S+)\s(\S+|-)\s(\S+)\s(\S+)\s(\w+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)\s(\S+)$/).each_with_index do |value, i|
+            next unless CloudfrontAccessLogParser.fields[i - 1]
+
+            mapping[CloudfrontAccessLogParser.fields[i - 1]] = value unless i.zero?
+            x = value if i.zero?
+          end
+          raise ArgumentError, "bad format: '#{line}, value #{x}'" unless x
+
+          mapping[:date] = Time.zone.parse(mapping[:date] + " " + mapping[:time] + " UTC") unless mapping[:date].nil?
+          mapping[:server_ip] = IP.new(mapping[:"x-forwarded-for"]) unless ["-", nil].include?(mapping[:"x-forwarded-for"])
+          mapping[:client_ip] = IP.new(mapping[:"c-ip"]) unless mapping[:"c-ip"].nil?
+
+          mapping[:port] = mapping[:"c-port"].to_i
+          mapping[:status] = mapping[:"sc-status"].to_i
+
+          mapping[:time_taken] = mapping[:"time-taken"].to_f / 1000
+
+          mapping[:query] = mapping[:"cs-uri-query"]
+          mapping[:query] = nil if mapping[:query] == "-"
```

Comment:
The treatment here differs from the treatment for `server_ip` above, as that key won't be set if the mapped value is `-`, whereas here the key will be set but the value for it will be `nil`.

Is that intentional?

Reply:
Nope 😳